### DISCUSSION TITLE
A basic question for beginners, how to use view

### DISCUSSION THREAD

**Discussion by greenking19** [simolus3/drift#1887] [Q&A]
i am a newbie
demo of this tutorial,
But I don't know how to use TodoCategoryItemCount. 
How to use this View?
Are there any relevant examples?
```dart
@DataClassName('TodoCategory')
class TodoCategories extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get name => text()();
}

class TodoItems extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get title => text()();
  TextColumn get content => text().nullable()();
  IntColumn get categoryId => integer().references(TodoCategories, #id)();

  TextColumn get generatedText => text().nullable().generatedAs(
      title + const Constant(' (') + content + const Constant(')'))();
}

abstract class TodoCategoryItemCount extends View {
  TodoItems get todoItems;
  TodoCategories get todoCategories;

  Expression<int> get itemCount => todoItems.id.count();

  @override
  Query as() => select([
    todoCategories.name,
    itemCount,
  ]).from(todoCategories).join([
    innerJoin(todoItems, todoItems.categoryId.equalsExp(todoCategories.id))
  ]);
}
````

**Comment by simolus3** [comment]
> How to use this View? How to use this View?

Is there anything specific that you're looking for? You can just select from a view as if it were a table. So here, it would probably be `select(todoCategoryItemCount).get()` to select all rows from the view for instance.

I agree that the documentation or examples on using views could be clearer, but if you're interested in something specific I can give better examples :D

**Reply by greenking19** [reply]
> > How to use this View? How to use this View?
> 
> Is there anything specific that you're looking for? You can just select from a view as if it were a table. So here, it would probably be `select(todoCategoryItemCount).get()` to select all rows from the view for instance.
> 
> I agree that the documentation or examples on using views could be clearer, but if you're interested in something specific I can give better examples :D

In the process of using the view, the only thing I get is null. This confuses me 🥲
I found the problem because my code didn't add the TodoCategories row.
It's a basic bug, but it also taught me a lot. 
Thank you for your reply
Thank you bro😎
